CODIGO/clsTexManager.cls

```diff
@@ -176,32 +202,25 @@ On Error GoTo ErrHandler
         With .SurfaceEntry(.surfaceCount)
             .FileName = FileNum
             .UltimoAcceso = GetTickCount()
-            #If Compresion = 1 Then
-                If Not Extract_File_To_Memory(Graphics, App.path & "\..\Recursos\OUTPUT\", LTrim(str(FileNum)) & ".png", bytArr, ResourcesPassword) Then
-                    Debug.Print "¡No se puede cargar el grafico numero " & FileNum & "!"
-                    Exit Function
-                End If
-                Set .Texture = mD3D.CreateTextureFromFileInMemoryEx(device, bytArr(0), UBound(bytArr) + 1, D3DX_DEFAULT, D3DX_DEFAULT, 1, 0, D3DFMT_A8R8G8B8, D3DPOOL_MANAGED, D3DX_FILTER_LINEAR, D3DX_FILTER_LINEAR, &HFF000000, ByVal 0, ByVal 0)
-            #Else
-                Dim PathToFile As String
-                PathToFile = App.path & "\..\Recursos\Graficos\" & LTrim(str(FileNum)) & ".png"
-                If Not FileExist(PathToFile, vbArchive) Then
-                    Debug.Print "¡No se puede cargar el grafico numero " & PathToFile & "!"
-                    Exit Function
-                End If
-
-                Set .Texture = mD3D.CreateTextureFromFileEx(device, PathToFile, 0, 0, _
-                     1, 0, D3DFMT_A8R8G8B8, D3DPOOL_MANAGED, D3DX_FILTER_LINEAR, D3DX_FILTER_LINEAR, &HFF000000, ByVal 0, ByVal 0)
-            #End If
+            LoadResult = LoadTexture(str(FileNum), .Texture)
+            If LoadResult = 7 Then
```

Comment:
Replace 7 by the corresponding error literal for example `E_OUTOFMEMORY` or `D3DERR_OUTOFVIDEOMEMORY` ?

see https://learn.microsoft.com/en-us/windows/win32/direct3d9/d3dxcreatetexturefromfileex